File: src/CorporateMessengerWithAgile/ReactApp/reactapp.client/src/models/entity/TeamDetailsDto.ts
// src/models/entity/TeamDetailsDto.ts
import { z } from 'zod';
import { BaseDtoSchema } from './BaseDto';
import { GuidSchema } from '../Guid';
import { EmployeeDtoSchema } from './EmployeeDto';
import { SprintDtoSchema } from './SprintDto';

export const TeamDetailsDtoSchema = BaseDtoSchema.extend({
    projectId: GuidSchema,
    title: z.string(),
    standardSprintDuration: z.number(),
    users: z.array(EmployeeDtoSchema).default([]),
    sprints: z.array(SprintDtoSchema).default([]),
    kanbanBoardColumnIds: z.array(GuidSchema).optional(),
});

export type TeamDetailsDto = z.infer<typeof TeamDetailsDtoSchema>;

File: src/CorporateMessengerWithAgile/Application/Dto/TeamDetailsDto.cs
namespace Application.Dto
{
    public class TeamDetailsDto : BaseDto
    {
        public Guid ProjectId { get; set; }
        public string Title { get; set; } = null!;
        public int StandardSprintDuration { get; set; }
        public IReadOnlyList<EmployeeDto> Users { get; set; } = [];
        public IReadOnlyList<SprintDto> Sprints { get; set; } = [];
        public IReadOnlyList<KanbanBoardColumnDto> KanbanBoardColumnIds { get; set; } = [];
    }
}
